### PR TITLE
Md 4644: created Autocomplete component and added different types

### PR DESCRIPTION
**Pull request by abhishekraj1007**

### CODE REVIEW

File: src/components/AutoComplete/index.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    asyncFunc: () => Promise<OptionType[]>;
+};
+
+const AutocompleteCreate = <T,>(props: AutoCompleteCreateProps<T>): React.ReactElement => {
+    const { id, label, color, variant, addontext, onChange } = props;
+
+    const [inputText, setInputText] = useState<OptionType | null>(null);
+
+    const handleOnChange = (event: React.SyntheticEvent, newValue: any) => {
+        if (typeof newValue === 'string') {
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:36]: let title = newValue;
if(typeof newValue === 'string){
 title={ title: newValue}
}

File: src/components/AutoComplete/index.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
+import React, { useState, useEffect } from 'react';
+import CircularProgress from '@mui/material/CircularProgress';
+import { TextField } from '../TextField/index';
+
+import { default as MUIAutocomplete, createFilterOptions } from '@mui/material/Autocomplete';
+
+interface OptionType {
+    title: string;
+    inputValue?: string;
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:9]: as discussed we can remove this

File: src/components/AutoComplete/index.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+        onChange(event, newValue);
+    };
+
+    const getOptionLabelHandler = (option: any): string => {
+        if (typeof option === 'string') {
+            return option;
+        }
+
+        if (option.inputValue) {
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:56]: remove this

File: src/components/AutoComplete/index.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            return option.inputValue;
+        }
+
+        return option.title;
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:60]:
```suggestion
        return option[displayLabelKeyName];
```

File: src/components/AutoComplete/index.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                title: newValue.inputValue
+            });
+        } else {
+            setInputText(newValue);
+        }
+
+        onChange(event, newValue);
+    };
+
+    const getOptionLabelHandler = (option: any): string => {
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:51]:
```suggestion
    const getOptionLabelHandler = (option:??): string => {
```

File: src/components/AutoComplete/index.tsx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        }
+
+        if (option.inputValue) {
+            return option.inputValue;
+        }
+
+        return option.title;
+    };
+
+    const filterOptionHandler = (options: any, params: any) => {
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:63]:
```suggestion
    const filterOptionHandler = (options:??, params: any) => {
```

File: src/components/AutoComplete/index.tsx [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        const filter = createFilterOptions();
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:64]: This we can keep outside and can create once

Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:64]:
```suggestion
        const muiFilterOptions = createFilterOptions();
```

File: src/components/AutoComplete/index.tsx [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        let filtered = filter(options, params),
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:65]:
```suggestion
        const filteredResult = muiFilterOptions(options, params),
```

File: src/components/AutoComplete/index.tsx [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+            isExisting: boolean | undefined;
+
+        const { inputValue } = params;
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:68]: remove

File: src/components/AutoComplete/index.tsx [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+        if (inputValue) {
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:70]: remove

File: src/components/AutoComplete/index.tsx [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+            return option.inputValue;
+        }
+
+        return option.title;
+    };
+
+    const filterOptionHandler = (options: any, params: any) => {
+        const filter = createFilterOptions();
+        let filtered = filter(options, params),
+            isExisting: boolean | undefined;
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:66]: remove this

File: src/components/AutoComplete/index.tsx (hunk 7 above)
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:64]: Remove this entire thing

File: src/components/AutoComplete/index.tsx [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+                });
+            }
+        }
+
+        if (inputText && inputText.title) return filtered.filter((item: any) => item.title.toLowerCase().startsWith(inputText.title.toLocaleLowerCase()));
+
+        return filtered;
+    };
+
+    const renderOptionHandler = (args: any, option: any) => <li {...args}>{option.title}</li>;
```
Comment (abhijit-parspec) [inline src/components/AutoComplete/index.tsx:86]: we can remove

**Review by abhijit-parspec — CHANGES_REQUESTED** [review]

File: src/components/AutoComplete/topfilm.ts [hunk 13]
```diff
@@ -0,0 +1,131 @@
+type TopFilms = {
```
[the remark this answers is not in the document]
Comment (abhishekraj1007) [reply src/components/AutoComplete/topfilm.ts:1]: done

**Pull request closed without merging**